### PR TITLE
Add a new argument '--expand' to 'az policy state list'

### PR DESCRIPTION
**Pull request by j5lim**
Add support for `az policy state list --expand PolicyEvaluationDetails` to query policy evaluation details on the resource.
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-policyinsights/azure/cli/command_modules/policyinsights/_validators.py [hunk 1]
```diff
@@ -14,3 +14,15 @@ def validate_resource(cmd, namespace):  # pylint: disable=unused-argument
                 raise CLIError('--namespace is required if --resource is not a resource ID.')
             if not namespace.resource_type:
                 raise CLIError('--resource-type is required if --resource is not a resource ID.')
+
+
+def validate_resourceScopeRequired(cmd, namespace):
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-policyinsights/azure/cli/command_modules/policyinsights/_validators.py]: Would `validate_resource` be better? Either way "resourceScopeRequired" has to be split to be snake style

File: src/command_modules/azure-cli-policyinsights/azure/cli/command_modules/policyinsights/_validators.py [hunk 2]
```diff
@@ -14,3 +14,15 @@ def validate_resource(cmd, namespace):  # pylint: disable=unused-argument
...
+    if not namespace.resource:
+        raise CLIError('--resource is required for the operation.')
+
+    validate_resource(cmd, namespace)
+
+
+def validate_expand(cmd, namespace):
+    if namespace.expand_clause is not None:
+        validate_resourceScopeRequired(cmd, namespace)
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-policyinsights/azure/cli/command_modules/policyinsights/_validators.py]: again, please  fix the naming

File: src/command_modules/azure-cli-policyinsights/azure/cli/command_modules/policyinsights/_validators.py (hunk 1 above)
Comment (j5lim) [reply src/command_modules/azure-cli-policyinsights/azure/cli/command_modules/policyinsights/_validators.py]: @yugangw-msft  Changed to `validate_resource_required`

**Comment by j5lim** [comment]
@yugangw-msft azcli-CI failed. Looks like it is not relevant to the changes. Can you please take a look? Thanks.

**Comment by marstr** [comment]
Looks like the ADO artifact upload API timed out. I've restarted the build, this shouldn't be a recurring issue.

**Pull request merged**